e2e/__tests__/tsIntegration.test.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as path from 'path';
+import {onNodeVersions} from '@jest/test-utils';
+import {cleanup, writeFiles} from '../Utils';
+import {getConfig} from '../runJest';
+
+const DIR = path.resolve(__dirname, '../ts-node-integration');
+
+beforeEach(() => cleanup(DIR));
+afterAll(() => cleanup(DIR));
+
+describe('when `Config` type is imported from "@jest/types"', () => {
+  test('works with object config exported from TS file', () => {
+    writeFiles(DIR, {
+      '__tests__/dummy.test.js': "test('dummy', () => expect(123).toBe(123));",
+      'jest.config.ts':
+        'import type {Config} from "@jest/types";' +
```

Comment:
use `` ` `` instead of manually concating (and losing newlines - on errors the message will be hard to read)